elasticsearch/values.yaml

```diff
@@ -18,6 +18,24 @@ minimumMasterNodes: 2
 
 esMajorVersion: 6
 
+# Settings for your nodes, clusters, and indices.
+settings:
+  # These settings become part of elasticsearch.yml on the node.
+  # It should be structured, not a text block.
+  node: {}
+    # Example, to enable Elasticsearch Security
+    #xpack.security.enabled: true
+
+  # These settings are published using the /_cluster/settings endpoint
+  # TODO(sissel): Not implemented yet.
+  cluster: {}
```

Comment:
Is there a specific reason to split these out into cluster and index? Rather than just having a generic way to set settings that need to be done via the API? I ask this because the two settings I'm currently setting having the paths `/_snapshot/default` and `_all/_settings` which would not be covered by these. Instead just having a generic way to specify the path and the payload you want feels like it would work nicely. 

```
settings:
- path: /_all/_settings
  data:
    settings:
      index.unassigned.node_left.delayed_timeout: 5m
```

Which would just generate a curl command with the json payload. (pseudo code)

```
curl -XPUT 'http://elasticsearch:9200/{{ path }}' -H "Content-Type: application/json" -d "{{ toJson data }}"
```

Reply:
I am +1 on what you suggest, though I think this would mean we should separate API-driven-settings from `elasticsearch.yml`. It would feel weird to mix them:

```
settings:
  node: 
    # elasticsearch.yml contents here
  /_all/settings:
    index.unassigned.node_left.delayed_timeout: 5m
  ...
```

For your example, you used a list-of-maps, but lists cannot be merged in Helm across charts, and having two `/_all/_settings` entries doesn't make sense to me. Could it be a map of `path: settings` as I have in this comment?

Reply:
Thinking more about this, yeah I definitely agree on paths, since these could be used to deploy a whole bunch of things I've had to deal with lately: watches, security roles, cluster settings, _all index settings, etc.